thoth/user_api/api_v1.py

```diff
@@ -399,6 +401,129 @@ def list_python_package_indexes():
     return graph.python_package_index_listing()
 
 
+def post_build(
+    log_info: dict,
+    base_image: str = None,
+    image: str = None,
+    debug: bool = False,
+    registry_user: str = None,
+    registry_password: str = None,
+    environment_type: str = None,
+    origin: str = None,
+    verify_tls: bool = True,
+    force: bool = False,
+    output_format: str = "json",
+    handler: str = None,
+    limit: int = None,
+):
+    """Run analysis on a build."""
+    response = {"base_image_analysis": {}, "output_image_analysis": {}, "buildlog_analysis": {}}
+    status = 202
+    if image:
+        # Run image analysis
+        image_analyze_response, image_analyze_status = post_analyze(
+            image=image,
+            debug=debug,
+            registry_user=registry_user,
+            registry_password=registry_password,
+            environment_type=environment_type,
+            origin=origin,
+            verify_tls=verify_tls,
+            force=force,
+        )
+        if image_analyze_status == 202:
+            response["output_image_analysis"] = image_analyze_response
+        else:
+            status = image_analyze_status
+
+    if base_image:
+        # Run base image analysis
+        base_image_analyze_response, base_image_analyze_status = post_analyze(
+            image=base_image,
+            debug=debug,
+            environment_type=environment_type,
+            origin=origin,
+            verify_tls=verify_tls,
+            force=force,
+        )
+        if base_image_analyze_status == 202:
+            response["base_image_analysis"] = base_image_analyze_response
+        else:
+            status = base_image_analyze_status
+
+    if log_info:
+        # attach image analysis details to build log
+        if image and image_analyze_response:
+            log_info["image_analysis_id"] = image_analyze_response["analysis_id"]
+        if base_image and base_image_analyze_response:
+            log_info["base_image_analysis_id"] = base_image_analyze_response["analysis_id"]
+
+        # Run build log analysis
+        buildlog_analyze_response, buildlog_analyze_status = post_buildlog_analyze(
+            log_info=log_info, output_format=output_format, handler=handler, limit=limit, force=force
+        )
+        if buildlog_analyze_status == 202:
+            response["buildlog_analysis"] = buildlog_analyze_response
+        else:
+            status = buildlog_analyze_status
+
+    return response, status
+
+
+def post_buildlog_analyze(
```

Comment:
Maybe supporting just one endpoint for builds as implemented in `post_build` would be good enough as you implemented it. Users can pick which parameters they will provide (any combination of base image, output image, build log).